components/cache/adapters/redis_adapter.rst

```diff
@@ -0,0 +1,127 @@
+
+
+This adapter expects a `Redis`_, `RedisArray`_, `RedisCluster`_, or `Predis`_ instance to be
+passed as the first parameter. A namespace and default cache lifetime can optionally be passed
+as the second and third parameters::
+
+    use Symfony\Component\Cache\Adapter\RedisAdapter;
+
+    $cache = new RedisAdapter(
+
+        // the object that stores a valid connection to your Redis system
+        \Redis $redisConnection,
+
+        // the string prefixed to the keys of the items stored in this cache
+        $namespace = '',
+
+        // the default lifetime (in seconds) for cache items that do not define their
+        // own lifetime, with a value 0 causing items to be stored indefinitely (i.e.
+        // until MemcachedAdapter::clear() is invoked or the server(s) are purged)
+        $defaultLifetime = 0
+    );
+
+Configure the Connection
+------------------------
+
+The :method:`Symfony\\Component\\Cache\\Adapter\\RedisAdapter::createConnection`
+helper method allows creating and configuring the Redis client class instance using a
+`Data Source Name (DSN)`_ or an array of DSNs::
+
+    use Symfony\Component\Cache\Adapter\RedisAdapter;
+
+    // pass a single DSN string to register a single server with the client
+    $client = RedisAdapter::createConnection(
+        'redis://localhost'
+    );
+
+    // pass an array of DSN strings to register multiple servers with the client
+    $client = RedisAdapter::createConnection(array(
+        'redis://10.0.0.100',
+        'redis://10.0.0.101',
+        'redis://10.0.0.102',
+        // etc...
+    ));
+
+The DSN can specify either an IP/host (and an optional port) or a socket path, as well as a user
+and password (for SASL authentication) and a weight (for multiple server prioritization).
+
+.. note::
+
+    A `Data Source Name (DSN)`_ for this adapter must use the following format.
+
+    .. code-block:: text
+
+        redis://[user:pass@][ip|host|socket[:port]][/db-index]
+
+
+Below are common examples of valid DSNs showing a combination of available values::
+
+    use Symfony\Component\Cache\Adapter\RedisAdapter;
+
+    $client = RedisAdapter::createConnection(array(
+
+        // host "my.server.com" and port "6379"
+        'redis://my.server.com:6379',
+
+        // host "my.server.com" and port "6379" and database index "20"
+        'redis://my.server.com:6379/20'
+
+        // host "localhost" and SASL use "rmf" and pass "abcdef"
+        'redis://rmf:abcdef@localhost'
+
+        // socket "/var/run/redis.sock" and SASL user "user1" and pass "bad-pass"
+        'redis://user1:bad-pass@/var/run/redis.sock'
+
+    ));
+
+
+Configure the Options
+---------------------
+
+The :method:`Symfony\\Component\\Cache\\Adapter\\RedisAdapter::createConnection` helper method
+also accepts an array of options as its second argument. The expected format is an associative
+array of ``key => value`` pairs representing option names and their respective values::
```

Comment:
Yeah, how do we want to handle the options for Redis, since it supports a few extension/user-land class implementations?